src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoTreeNodes.cs

```diff
@@ -1644,6 +1649,35 @@ internal partial class ColumnsChildFactory : SmoChildFactoryBase
     {
         public override IEnumerable<string> ApplicableParents() { return new[] { "Columns" }; }
 
+        public override IEnumerable<INodeFilter> Filters
+        {
+            get
+            {
+                var filters = new List<INodeFilter>();
+                filters.Add(new NodePropertyFilter
+                {
+                    Property = "IsDroppedLedgerColumn",
+                    Type = typeof(bool),
+                    ValidFor = ValidForFlag.Sql2022|ValidForFlag.AzureV12,
+                    Values = new List<object> { 0 },
+                });
+                return filters;
+            }
+        }
+
+        protected override void OnExpandPopulateFolders(IList<TreeNode> currentChildren, TreeNode parent)
+        {
+            currentChildren.Add(new FolderNode {
```

Comment:
Couldn't you just check the parent here and only add the folder if the parent is a ledger table?

Reply:
Maybe, is there precedent for something like that? This is the auto-generated code from the SmoTreeNodesDefinition.

Reply:
Not that I'm aware, this is pretty unprecedented territory for ADS unfortunately =)

This being here is problematic though, what I'd say then is see if there's somewhere else that you can do the check instead that still keeps the same result (maybe on where this is called?)

Reply:
That's where I made the change, in OnExpandPopulateFoldersAndFilter :)